### PR TITLE
Deduplicate twig data creation for problems action.

### PR DESCRIPTION
**Pull request by meisterT**
No functional changes intended.

### CODE REVIEW

File: webapp/src/Controller/PublicController.php [hunk 1]
```diff
@@ -181,42 +181,8 @@ public function teamAction(Request $request, int $teamId)
...
-        }
-
-        return $this->render('public/problems.html.twig', [
-            'problems' => $problems,
-            'showLimits' => $showLimits,
-            'defaultMemoryLimit' => $defaultMemoryLimit,
-            'timeFactorDiffers' => $timeFactorDiffers,
-        ]);
+        return $this->render('public/problems.html.twig',
+            $this->dj->getTwigDataForProblemsAction(-1));
```
Comment (eldering) [inline webapp/src/Controller/PublicController.php:185]: This `-1` argument seems a bit hacky. Can't we make `$teamid` an optional argument, or use `null` instead?

File: webapp/src/Service/DOMJudgeService.php [hunk 2]
```diff
@@ -832,4 +832,43 @@ public function getContestStats(Contest $contest): array
             ->getSingleScalarResult();
         return $stats;
     }
+
+    public function getTwigDataForProblemsAction(int $userid): array {
```
Comment (eldering) [inline webapp/src/Service/DOMJudgeService.php:836]: `$userid` should be understood as `$teamid`, right?

Comment (meisterT) [reply webapp/src/Service/DOMJudgeService.php]: thanks, done

File: webapp/src/Controller/PublicController.php (hunk 1 above)
Comment (meisterT) [reply webapp/src/Controller/PublicController.php:185]: We might want to also clean up getCurrentContest(s), but I think it's separate from this cleanup.

See
https://github.com/DOMjudge/domjudge/blob/master/webapp/src/Service/DOMJudgeService.php#L180-L181 for the `-1` case, but `null` is also allowed - I am not sure if we use that in some places.

**Pull request merged**